datasquare.go

```diff
@@ -1,242 +1,248 @@
-    for i := uint(0); i < uint(len(newColumn)); i++ {
-        if len(newColumn[i]) != int(ds.chunkSize) {
-            return errors.New("invalid chunk size")
-        }
-    }
+	for i := uint(0); i < uint(len(newColumn)); i++ {
+		if len(newColumn[i]) != int(ds.chunkSize) {
+			return errors.New("invalid chunk size")
+		}
+	}
 
-    for i := uint(0); i < uint(len(newColumn)); i++ {
-        ds.square[x+i][y] = newColumn[i]
-    }
+	for i := uint(0); i < uint(len(newColumn)); i++ {
+		ds.square[x+i][y] = newColumn[i]
+	}
 
-    ds.resetRoots()
+	ds.resetRoots()
 
-    return nil
+	return nil
 }
 
 func (ds *dataSquare) resetRoots() {
-    ds.rowRoots = nil
-    ds.columnRoots = nil
+	ds.rowRoots = nil
+	ds.columnRoots = nil
 }
 
 func (ds *dataSquare) computeRoots() {
-    rowRoots := make([][]byte, ds.width)
-    columnRoots := make([][]byte, ds.width)
-    var rowTree *merkletree.Tree
-    var columnTree *merkletree.Tree
-    var rowData [][]byte
-    var columnData [][]byte
-    for i := uint(0); i < ds.width; i++ {
-        rowTree = merkletree.New(ds.hasher)
-        columnTree = merkletree.New(ds.hasher)
-        rowData = ds.Row(i)
-        columnData = ds.Column(i)
-        for j := uint(0); j < ds.width; j++ {
-            rowTree.Push(rowData[j])
-            columnTree.Push(columnData[j])
-        }
-
-        rowRoots[i] = rowTree.Root()
-        columnRoots[i] = columnTree.Root()
-    }
-
-    ds.rowRoots = rowRoots
-    ds.columnRoots = columnRoots
+	rowRoots := make([][]byte, ds.width)
+	columnRoots := make([][]byte, ds.width)
+	var rowTree *merkletree.Tree
+	var columnTree *merkletree.Tree
+	var rowData [][]byte
+	var columnData [][]byte
+	for i := uint(0); i < ds.width; i++ {
+		rowTree = merkletree.New(ds.hasher)
+		columnTree = merkletree.New(ds.hasher)
+		rowData = ds.Row(i)
+		columnData = ds.Column(i)
+		for j := uint(0); j < ds.width; j++ {
+			rowTree.Push(rowData[j])
+			columnTree.Push(columnData[j])
+		}
+
+		rowRoots[i] = rowTree.Root()
+		columnRoots[i] = columnTree.Root()
+	}
+
+	ds.rowRoots = rowRoots
+	ds.columnRoots = columnRoots
 }
 
 // RowRoots returns the Merkle roots of all the rows in the square.
 func (ds *dataSquare) RowRoots() [][]byte {
-    if ds.rowRoots == nil {
-        ds.computeRoots()
-    }
+	if ds.rowRoots == nil {
+		ds.computeRoots()
+	}
 
-    return ds.rowRoots
+	return ds.rowRoots
 }
 
 // ColumnRoots returns the Merkle roots of all the columns in the square.
 func (ds *dataSquare) ColumnRoots() [][]byte {
-    if ds.columnRoots == nil {
-        ds.computeRoots()
-    }
+	if ds.columnRoots == nil {
+		ds.computeRoots()
+	}
 
-    return ds.columnRoots
+	return ds.columnRoots
 }
 
 func (ds *dataSquare) computeRowProof(x uint, y uint) ([]byte, [][]byte, uint, uint) {
-    tree := merkletree.New(ds.hasher)
-    tree.SetIndex(uint64(y))
-    data := ds.Row(x)
+	tree := merkletree.New(ds.hasher)
+	err := tree.SetIndex(uint64(y))
+	if err != nil {
+		panic(err)
```

Comment:
Shouldn't we pass the error upstream if an invalid tree index is supplied?

Reply:
Yes, absolutely! Good catch. All other places were in tests where it was OK to panic I guess. Will update in a bit.